device_backends/NumericAddressedBackend/src/NumericAddressedBackend.cc

```diff
@@ -249,7 +266,11 @@ namespace ChimeraTK {
     // This function just makes sure that at() is used to access the _primaryInterruptDistributors map,
     // which guarantees that the map is not altered.
     VersionNumber v{};
-    _primaryInterruptDistributors.at(interruptNumber)->distribute(nullptr, v);
+    auto asyncDomain = _asyncDomainImpls.at(interruptNumber)->lock();
```

Comment:
It has to be ```at()``` because ```[ ]``` modifies the container, but the container is const. You can iterate a const map from multiple threads without a lock, which we do for efficiency here.

The thread for interrupt N which is calling ```dispatchInterrupt(N)"" has been started by something inside ```_primaryInterruptDistributors.at(N)```, and the _primaryInterruptDistributors are created in the constructor and never go away. So we don't have to check that it is throwing here.